staging/src/k8s.io/apimachinery/pkg/version/helpers_test.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"testing"
+)
+
+func TestCompareKubeAwareVersionStrings(t *testing.T) {
+	tests := []*struct {
+		v1, v2          string
+		expectedGreater bool
+	}{
+		{"v1", "v2", false},
+		{"v2", "v1", true},
+		{"v10", "v2", true},
+		{"v1", "v2alpha1", true},
+		{"v1", "v2beta1", true},
+		{"v1alpha2", "v1alpha1", true},
+		{"v1beta1", "v2alpha3", true},
+		{"v1alpha10", "v1alpha2", true},
+		{"v1beta10", "v1beta2", true},
```

Comment:
Add cases comparing kubversion against unformatted string and two unformatted strings please.